MediaBrowser.Model/Dlna/StreamInfo.cs

```diff
@@ -618,201 +620,296 @@ public string GetOption(string name)
             {
-                list.Add(new NameValuePair("StartTimeTicks", string.Empty));
+                if (StartPositionTicks != 0)
+                {
+                    sb.Append("&StartTimeTicks=");
+                    sb.Append(StartPositionTicks.ToString(CultureInfo.InvariantCulture));
+                }
             }
             else
             {
-                list.Add(new NameValuePair("StartTimeTicks", startPositionTicks.ToString(CultureInfo.InvariantCulture)));
+                if (!string.IsNullOrEmpty(Container))
+                {
+                    sb.Append("&SegmentContainer=");
+                    sb.Append(Container);
+                }
+
+                if (SegmentLength.HasValue)
+                {
+                    sb.Append("&SegmentLength=");
+                    sb.Append(SegmentLength.Value.ToString(CultureInfo.InvariantCulture));
+                }
+
+                if (MinSegments.HasValue)
+                {
+                    sb.Append("&MinSegments=");
+                    sb.Append(MinSegments.Value.ToString(CultureInfo.InvariantCulture));
+                }
+
+                sb.Append("&BreakOnNonKeyFrames=");
+                sb.Append(BreakOnNonKeyFrames.ToString(CultureInfo.InvariantCulture));
             }
 
-            list.Add(new NameValuePair("PlaySessionId", item.PlaySessionId ?? string.Empty));
-            list.Add(new NameValuePair("api_key", accessToken ?? string.Empty));
+            if (!string.IsNullOrEmpty(PlaySessionId))
+            {
+                sb.Append("&PlaySessionId=");
+                sb.Append(PlaySessionId);
+            }
 
-            string liveStreamId = item.MediaSource?.LiveStreamId;
-            list.Add(new NameValuePair("LiveStreamId", liveStreamId ?? string.Empty));
+            if (!string.IsNullOrEmpty(accessToken))
+            {
+                sb.Append("&api_key=");
+                sb.Append(accessToken);
+            }
 
-            list.Add(new NameValuePair("SubtitleMethod", item.SubtitleStreamIndex.HasValue && item.SubtitleDeliveryMethod != SubtitleDeliveryMethod.External ? item.SubtitleDeliveryMethod.ToString() : string.Empty));
+            var liveStreamId = MediaSource?.LiveStreamId;
+            if (!string.IsNullOrEmpty(liveStreamId))
+            {
+                sb.Append("&LiveStreamId=");
+                sb.Append(liveStreamId);
+            }
 
-            if (!item.IsDirectStream)
+            if (!IsDirectStream)
             {
-                if (item.RequireNonAnamorphic)
+                if (RequireNonAnamorphic)
                 {
-                    list.Add(new NameValuePair("RequireNonAnamorphic", item.RequireNonAnamorphic.ToString(CultureInfo.InvariantCulture).ToLowerInvariant()));
+                    sb.Append("&RequireNonAnamorphic=");
+                    sb.Append(RequireNonAnamorphic.ToString(CultureInfo.InvariantCulture).ToLowerInvariant());
```

Comment:
```suggestion
                    sb.Append(RequireNonAnamorphic.ToString(CultureInfo.InvariantCulture));
```

The `ToLowerInvariant` isn't required, the api is case-insensitive

Reply:
@Shadowghost I should have specified- the `ToLowerInvariant()` is used in multiple places in the builder- none of which are required

Reply:
Should be fixed now. Tests required some fixing too.